src/main/scala/cromwell/backend/model/TaskStatus.scala

```diff
@@ -1,21 +1,24 @@
 package cromwell.backend.model
 
-import cromwell.backend.model.Status.Status
+import cromwell.caching.ExecutionHash
+import wdl4s.values.WdlValue
 
-/**
-  * Represents status of an execution.
-  */
-object Status extends Enumeration {
-  type Status = Value
-  val Created, Running, Succeeded, Failed, Canceled = Value
-}
+import scala.concurrent.Future
 
-/**
-  * Defines a task intermediate status.
-  */
-case class TaskStatus(status: Status) extends ExecutionEvent
 
-/**
-  * Defines a task final status with the resulting data.
-  */
-case class TaskFinalStatus(status: Status, result: ExecutionResult) extends ExecutionEvent
\ No newline at end of file
+sealed trait TaskStatus extends ExecutionEvent
+
+sealed trait IntermediateTaskStatus extends TaskStatus
+final case class CreatedTaskStatus(stdout: String, stderr: String) extends IntermediateTaskStatus
+case object RunningTaskStatus extends IntermediateTaskStatus
+
+sealed trait FinalTaskStatus extends TaskStatus
+final case class SucceededTaskStatus(outputs: Map[String, WdlValue], returnCode: Int, hash: ExecutionHash) extends FinalTaskStatus
+final case class FailedTaskStatus(error: Throwable, returnCode: Option[Int]) extends FinalTaskStatus
```

Comment:
I can remember we removed the option from case classes as per request of @geoffjentry. Shouldn't it apply here as well? So it can be represented in two different objects?